src/v/cloud_topics/housekeeper/housekeeper.cc

```diff
@@ -0,0 +1,127 @@
+/*
```

Comment:
> Note that intentionally are a little lax with bytes based retention. We
don't account for L0 data, and we allow going over the limit a bit (this
is implemented in the metastore, but still).

it's probably just terminology, but when you say "we allow going over the limit a bit" do you mean (1) we remove more data than X or (2) we remove less data than X, where X would be the precise amount of data that could be removed according to the time or size retention if we had byte-level granularity control?

Reply:
(2)!